gusto/diagnostics.py

```diff
@@ -1720,4 +1726,36 @@ def setup(self, domain, state_fields):
         m_X = state_fields(self.mixing_ratio_name)
         rho_d = state_fields(self.density_name)
         self.expr = m_X*rho_d
-        super().setup(domain, state_fields)
+
+        if self.space is None:
```

Comment:
Final thing: could you add some comments to explain why we want to have a specific space for the tracer density? Thanks!

Reply:
Sure, I've added a few more comments.